src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.FileSystem.cs

```diff
@@ -0,0 +1,462 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // Contains test methods that can be used for FileInfo, DirectoryInfo, File or Directory.
+    public abstract class BaseSymbolicLinks_FileSystem : BaseSymbolicLinks
+    {
+        private const string ExtendedPrefix = @"\\?\";
+        /// <summary>Creates a new file or directory depending on the implementing class.
+        /// If createOpposite is true, creates a directory if the implementing class is for File or FileInfo, or
+        /// creates a file if the implementing class is for Directory or DirectoryInfo.</summary>
+        protected abstract void CreateFileOrDirectory(string path, bool createOpposite = false);
+        protected abstract void AssertIsCorrectTypeAndDirectoryAttribute(FileSystemInfo linkInfo);
+        protected abstract void AssertLinkExists(FileSystemInfo linkInfo);
+        /// <summary>Calls the actual public API for creating a symbolic link.</summary>
+        protected abstract FileSystemInfo CreateSymbolicLink(string path, string pathToTarget);
+        /// <summary>Calls the actual public API for resolving the symbolic link target.</summary>
+        protected abstract FileSystemInfo ResolveLinkTarget(string linkPath, bool returnFinalTarget = false);
+
+        /// <summary>
+        /// Verifies that FileSystemInfo.LinkTarget matches the specified expected path.
+        /// If the current platform is Windows and <paramref name="actual"/> is absolute, this method asserts that <paramref name="actual"/> starts with \\?\.
+        /// </summary>
+        protected static void AssertLinkTargetEquals(string expected, string actual)
+        {
+#if WINDOWS
+            if (Path.IsPathFullyQualified(actual))
+            {
+                actual.StartsWith(ExtendedPrefix);
```

Comment:
if we switch from DeviceIoControl to NtQueryInformationFile , can we get rid of this logic?

Reply:
Can you please elaborate on this, for my own education? Why should we switch to NtQueryInformationFile?